plugin-locationlayer/src/main/java/com/mapbox/mapboxsdk/plugins/locationlayer/LocationLayerPlugin.java

```diff
@@ -519,6 +519,28 @@ public LocationEngine getLocationEngine() {
     return locationEngine;
   }
 
+  /**
+   * Sets the compass engine used to provide compass heading values.
+   *
+   * @param compassEngine to be used
+   * @since 0.8.0
+   */
+  public void setCompassEngine(@NonNull CompassEngine compassEngine) {
+    this.compassEngine = compassEngine;
```

Comment:
I think we should remove the listener attached to the current `CompassEngine` before swapping it.